app/models/build.rb

```diff
@@ -17,13 +17,18 @@ class Build < ActiveRecord::Base
   validates :project, presence: true
   validates :git_sha, allow_nil: true, format: SHA1_REGEX
   validates :dockerfile, presence: true, unless: :external?
-  [:dockerfile, :image_name].each do |attribute|
-    validates(
-      :git_sha,
-      allow_nil: true,
-      uniqueness: {scope: attribute, message: "already exists with this #{attribute}"},
-      if: attribute
-    )
+  [:git_sha, :external_url].each do |attribute|
+    [:dockerfile, :image_name].each do |scope|
```

Comment:
that's already merged ? <-> rebase ?

Reply:
#3510 isn't merged yet, but yes they are the same commits, will disappear once merged.